plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/TransformationMatrix.cpp

```diff
@@ -1263,7 +1309,12 @@ MStatus TransformationMatrix::translateTo(const MVector& vector, MSpace::Space s
     {
       insertTranslateOp();
     }
-    pushTranslateToPrim();
+
+    // Push new value to prim, but only if it's changing, otherwise extra work and unintended
+    // side effects will happen.
+    //
+    if (!apxEq(vector, m_translationFromUsd))
```

Comment:
First - as noted above, perhaps we should use vector::isEquivalent here.  I also think that it's default (1e-10) is a little safer, since a value of 1e-4 could easily be noticeable, especially if there's a scene that has smaller-than-average scale.  (Shouldn't happen, but sometimes does.)  If 1e-10 is still letting through changes essentially just due to floating point roundoff error to sneak through... maybe we could try 1e-8?

Also, you're not accounting for the tweak value here.  I guess simplest way to get the same behavior as current - while avoiding unnecessary over creation - would be something like:

```
if(apEx(..))
{
  m_translationTweak = MVector(0, 0, 0);
}
else
{
  pushTranslateToPrim();
}
```

Reply:
I'm happy to swap my non-standard approximate equality checking for some more standard function - I didn't know what was available and figured that if there was something better someone would comment and point me at the better way.
Regarding your tweak comment, could you please elaborate on what you mean?  Are you saying that I need to set tweak to zero if not updating m_translationFromUsd?  Wouldn't it already be correctly set in this case?

Reply:
I didn't do anything with the tweak because no one explained why resetting it was necessary here.  Will be happy to revisit.